delta-lake/src/main/java/org/apache/iceberg/delta/BaseMigrateDeltaLakeTableAction.java

```diff
@@ -0,0 +1,364 @@
+      AddFile addFile = (AddFile) action;
+      path = addFile.getPath();
+      partitionValues = addFile.getPartitionValues();
+    } else if (action instanceof RemoveFile) {
+      RemoveFile removeFile = (RemoveFile) action;
+      path = removeFile.getPath();
+      partitionValues = removeFile.getPartitionValues();
+    } else {
+      throw new IllegalStateException(
+          String.format(
+              "Unexpected action type for Delta Lake: %s", action.getClass().getSimpleName()));
+    }
+
+    String fullFilePath = getFullFilePath(path);
+
+    if (partitionValues == null) {
+      // For unpartitioned table, the partitionValues should be an empty map rather than null
+      throw new ValidationException("File %s does not specify a partitionValues", fullFilePath);
+    }
+
+    FileFormat format = determineFileFormatFromPath(fullFilePath);
+    FileIO io = table.io();
+    InputFile file;
+    if (io != null) {
+      file = io.newInputFile(fullFilePath);
+    } else {
+      file = HadoopInputFile.fromPath(new Path(fullFilePath), this.hadoopConfiguration);
+    }
+
+    MetricsConfig metricsConfig = MetricsConfig.forTable(table);
+    String nameMappingString = table.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+    NameMapping nameMapping =
+        nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+    Metrics metrics = getMetricsForFile(file, format, metricsConfig, nameMapping);
+
+    String partition =
+        spec.fields().stream()
+            .map(PartitionField::name)
+            .map(name -> String.format("%s=%s", name, partitionValues.get(name)))
+            .collect(Collectors.joining("/"));
+
+    return DataFiles.builder(spec)
+        .withPath(fullFilePath)
+        .withFormat(format)
+        .withFileSizeInBytes(file.getLength())
+        .withMetrics(metrics)
+        .withPartitionPath(partition)
+        .build();
+  }
+
+  private FileFormat determineFileFormatFromPath(String path) {
+    if (path.endsWith(PARQUET_SUFFIX)) {
+      return FileFormat.PARQUET;
+    } else if (path.endsWith(AVRO_SUFFIX)) {
+      return FileFormat.AVRO;
+    } else if (path.endsWith(ORC_SUFFIX)) {
+      return FileFormat.ORC;
+    } else {
+      throw new ValidationException("The format of the file %s is unsupported", path);
+    }
+  }
+
+  private Metrics getMetricsForFile(
+      InputFile file, FileFormat format, MetricsConfig metricsSpec, NameMapping mapping) {
+    try {
+      switch (format) {
+        case AVRO:
+          long rowCount = Avro.rowCount(file);
+          return new Metrics(rowCount, null, null, null, null);
+        case PARQUET:
+          return ParquetUtil.fileMetrics(file, metricsSpec, mapping);
+        case ORC:
+          return OrcMetrics.fromInputFile(file, metricsSpec, mapping);
+        default:
+          throw new ValidationException("Unsupported file format: %s", format);
+      }
+    } catch (UncheckedIOException e) {
```

Comment:
why do we need this try catch? It's already `UncheckedIOException` which is a runtime exception